id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java

```diff
@@ -289,14 +289,19 @@ public Map<AllowedKycDto, Object> prepareSharableAttributes(IdResponseDTO idResp
 			Object formattedObject=object;
 			
 				if (object != null) {
-					if(userReqMaskingAttributes!=null && userReqMaskingAttributes.contains(attribute)) {
-						 formattedObject=maskData(object.toString());
-					 }else if(userReqFormatingAttributes != null && userReqFormatingAttributes.containsKey(attribute)) {
-						 formattedObject = filterAndFormat(key,identity,userReqFormatingAttributes);
-					 }
+					if (userReqMaskingAttributes != null && !userReqMaskingAttributes.isEmpty()
+							&& userReqMaskingAttributes.contains(attribute)) {
+						formattedObject = maskData(object.toString());
+					} 
+					if (userReqFormatingAttributes != null && !userReqFormatingAttributes.isEmpty()
+							&& userReqFormatingAttributes.containsKey(attribute)) {
+						formattedObject = filterAndFormat(key, identity, userReqFormatingAttributes);
+					}
 					attributesMap.put(key, formattedObject);
-				} else {
-					 if (attribute.equalsIgnoreCase(CredentialConstants.ENCRYPTIONKEY)) {
+				} else if (attribute.equalsIgnoreCase(CredentialConstants.FULLNAME)) {
```

Comment:
@ramadeviidupulapati , what is this condition for?
`(attribute.equalsIgnoreCase(CredentialConstants.FULLNAME))` ?